vignettes/adrs_gcig.Rmd

```diff
@@ -0,0 +1,799 @@
+  ~MCRIT1ML, ~MCRIT1MN,
+  "PDCA125", CA2XULRR == "Y" & CANORM2X == "Y", 
+  "Patients with elevated CA-125 before treatment and normalization of CA-125 (A)", 1,
+  "PDCA125", CA2XULRR == "Y" & CNOTNORM == "Y", 
+  "Patients with elevated CA-125 before treatment, which never normalizes (B)", 2,
+  "PDCA125", CA2XULRR == "N" & CANORM2X == "Y", 
+  "Patients with CA-125 in the reference range before treatment (C)", 3
+)
+
+adrs <- adrs %>%
+  mutate(MCRIT1 = ifelse(PARAMCD == "PDCA125", "PD  Category Group", NA_character_)) %>%
+  derive_vars_cat(
+    definition = definition_mcrit,
+    by_vars = exprs(PARAMCD)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVALC, MCRIT1, MCRIT1ML, MCRIT1MN),
+  filter = (PARAMCD == "PDCA125" & !is.na(AVALC))
+)
+```
+
+Derivation of the progression category may be more complex if the data is collected in a different way and the user needs to check whether:
+
+- baseline CA-125 level was elevated,
+- CA-125 level normalizes over the course of the study,
+- CA-125 level > 2 * ULRR (if CA-125 level normalizes) or CA-125 level > 2 * nadir (if CA-125 level never normalizes).
+
+If criterion to find PD category draws from multiple rows (different parameters or multiple rows for a single parameter) this will require the creation of a new parameter.
+
+## CA-125 Best Confirmed Overall Response {#cborca}
+
+The function `admiral::derive_extreme_event()` can be used to derive the CA-125 Best Confirmed Overall Response Parameter.
+
+Some events such as `bor_cr`, `bor_pr` have been defined in {admiralonco}. Missing events specific to GCIG criteria are defined below.  
+**Note**: *For `SD`, it is not required as for RECIST1.1 that the response occurs after a protocol-defined number of days.*
+
+```{r}
+ovr <- ovr_ca125
+
+bor_sd_gcig <- event(
+  description = "Define stable disease (SD) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "SD",
+  set_values_to = exprs(AVALC = "SD")
+)
+
+bor_ne_gcig <- event(
+  description = "Define not evaluable (NE) for best overall response (BOR)",
+  dataset_name = "ovr",
+  condition = AVALC == "NE",
+  set_values_to = exprs(AVALC = "NE")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = get_admiral_option("subject_keys"),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr, ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(
+      bor_cr, bor_pr, bor_sd_gcig, bor_pd, bor_ne_gcig, no_data_missing
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CBORCA",
+      PARAM = "CA-125 Best Confirmed Overall Response by Investigator",
+      PARAMN = 5,
+      PARCAT1 = "Investigator",
+      PARCAT1N = 1,
+      PARCAT2 = "CA-125",
+      PARCAT2N = 1,
+      AVAL = aval_resp(AVALC),
+      ANL01FL = "Y",
+      ANL02FL = "Y"
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
```

Comment:
for 701-1015 subj why we have CBORCA should be PD, but its SD. not sure why?
Also we have 8 subjects in the data, we get only 7 subj with CBORCA assigned?
can we also sort the data by subject id?

Reply:
Why PD? For this subjects I see responses SD/NE/PR independent of response category. 
I see 8 subjects with not MISSING response.
In my case it is sorted by “subject_keys”, PARAMN, ADT although I did not call the sort command additionally.

Reply:
@starosto for 701-1015 you are right as this is only best CA125 response. SD looks correct. Not sure I see only 7 subjects in the outputs, but you see 8 its fine we can close this issue.